internal/driver/glfw/window.go

```diff
@@ -256,21 +252,29 @@ func (w *window) fitContent() {
 	minWidth, minHeight := w.minSizeOnScreen()
 	w.viewLock.Lock()
 	defer w.viewLock.Unlock()
-	if w.width < minWidth || w.height < minHeight {
-		if w.width < minWidth {
-			w.width = minWidth
-		}
-		if w.height < minHeight {
-			w.height = minHeight
-		}
-		w.viewport.SetSize(w.width, w.height)
-	}
 	if w.fixedSize {
-		w.width = internal.ScaleInt(w.canvas, w.Canvas().Size().Width)
-		w.height = internal.ScaleInt(w.canvas, w.Canvas().Size().Height)
+		limitSize := internal.ScaleSize(w.canvas, w.Canvas().Size())
+		if limitSize.Width == 0 || limitSize.Height == 0 {
+			// When the content hasn't been set yet, the min size is not available yet and
+			// their is no point into doing anything
+			return
+		}
 
-		w.viewport.SetSizeLimits(w.width, w.height, w.width, w.height)
+		w.viewport.SetSizeLimits(limitSize.Width, limitSize.Height, limitSize.Width, limitSize.Height)
+		w.viewport.SetSize(limitSize.Width, limitSize.Height)
 	} else {
+		// Adjust size to be at least minSize
+		if w.size.Width < minWidth || w.size.Height < minHeight {
```

Comment:
This is no longer applying to FixedSize - is it your intent to allow FixedSize Resize() to be less than MinSize()?

Reply:
I believe it is, but in an indirect way. When canvas.SetContent is called, it resize the window to the new canvas content.MinSize. So at this point, I think the canvas size is already constrained to be MinSize().

Reply:
Are you sure? Really we have instilled the minSize for window in the non-fixed scenario as well, so if constraints are respected it should not be needed in any instance.
Think about it programatically. If someone calls Resize() manually on a fixed size window this will now allow it to be smaller than minSize...

Reply:
Hum, I have build this with the assumption that calling Resize on a fixed size window once it has a content would be a noop if the new requested size was below the SizeLimits. I guess I should check that before calling viewport.SetSize in case the window manager doesn't enforce it when coming from the application. I have also realized that the call to fitContent in Resize is unnecessary.